routing/dht/routing.go

```diff
@@ -28,6 +29,8 @@ var asyncQueryBuffer = 10
 // PutValue adds value corresponding to given Key.
 // This is the top level "Store" operation of the DHT
 func (dht *IpfsDHT) PutValue(ctx context.Context, key key.Key, value []byte) error {
+	ctx, cancel := context.WithTimeout(ctx, time.Second*10) // CR: should this be removed instead?
```

Comment:
i think it makes sense to have timeouts on dht ops. but 10s may be low. very bad for high latency links. maybe 30s? 1m?